core/modules/entity/entity.module

```diff
@@ -43,6 +43,33 @@ function entity_flush_caches() {
   return $bins;
 }
 
+/**
+ * Implements hook_theme().
+ */
+function entity_theme() {
+  // Build a pattern in the form of "(type1|type2|...)(\.|__)" such that all
+  // templates starting with an entity type or named like the entity type
+  // are found.
+  // This has to match the template suggestions provided in
+  // template_preprocess_entity().
+  $types = array();
+  foreach (entity_get_info() as $type => $info) {
```

Comment:
The following lines of code are causing a PHP Fatal Error when doing a "clean" install of a patched Backdrop with this PR. While I don't have an exact cause, I believe the issue is that this line is calling `entity_get_info` before all the necessary inc files have been loaded (this is happening during the first stages of the install cycle).

I'm not sure why we haven't experienced this Fatal Error with the Tugboat test sites. Perhaps Tugboat doesn't do a clean install from scratch of Backdrop? Or perhaps it applies these patches after installing? (which doesn't cause the error).  

I'm haven't come up with a solution for this yet. These lines are needed so that we have access to the appropriate template files for custom entities "out of the box" . Perhaps this `pattern` can be inserted in  an implementation of `hook_theme_registry_alter`? I haven't tested this approach yet. Will report after I do.

Reply:
Do you mind explaining how you make the patched Backdrop?

Reply:
No problem. I download Backdrop 1.x from the repository, then download this PR as a patch here: https://patch-diff.githubusercontent.com/raw/backdrop/backdrop/pull/3848.patch

Then I apply the patch BEFORE installing Backdrop running a `patch` command in shell. Then install the site normally from the browser

Reply:
Youre right, fatal error confirmed. On loading `install.core.php` Backdrop loads modules User, Entity and System. Now that Entity calls `entity_get_info()` immediately, this calls `user_entity_info()` which calls `db_table_exists()` which doesnt yet exist because the `database.inc` file isnt yet loaded.

I can solve this by wrapping that code in 
```
  global $databases;
  if (!empty($databases)) {
    foreach (entity_get_info() as $type => $info) {
    ... etc
```
Or we could probably check bootstrapping level.